t/pgtap/first_release_dates.sql

```diff
@@ -225,6 +225,149 @@ SELECT results_eq(
   'VALUES (1970::smallint, 11::smallint, null::smallint)'
 );
 
+UPDATE release SET status = 6 WHERE id = 3;
+
+SELECT results_eq(
+  'all_release_first_release_dates',
+  $$
+    VALUES
+      (1::integer, 1999::smallint, 1::smallint, 4::smallint),
+      (2::integer, 1989::smallint, null::smallint, null::smallint)
+  $$,
+  'Release dates as expected after marking release as cancelled'
+);
+
+SELECT results_eq(
+  'release_group_first_release_dates',
+  $$
+    VALUES
+      (1::integer, 1999::smallint, 1::smallint, 4::smallint),
+      (2::integer, 1989::smallint, null::smallint, null::smallint)
+  $$,
+  'RG dates as expected after marking release as cancelled'
+);
+
+SELECT results_eq(
+  'recording_1_first_release_date',
+  'VALUES (1989::smallint, null::smallint, null::smallint)',
+  'Recording date as expected after marking release as cancelled'
```

Comment:
To clarify it is not about any other date (such as for recording relationships):
```suggestion
  'Recording (first release) date as expected after marking release as cancelled'
```